### PR TITLE
Update scala.js to 1.0.0

### PR DESCRIPTION
**Pull request by ybasket**
As this Scala.js 0.6 and 1.0 aren't compatible, this is important for libraries/projects depending on scala-newtype.

Includes:
* update sbt plugins
* update cats versions (disabling cats tests on JS)
* update test libs, handle deprecations

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -15,18 +15,23 @@ lazy val newtype = crossProject(JSPlatform, JVMPlatform).in(file("."))
 lazy val newtypeJVM = newtype.jvm
 lazy val newtypeJS = newtype.js
 
-lazy val catsTests = crossProject(JSPlatform, JVMPlatform).in(file("cats-tests"))
```
Comment (ybasket) [inline build.sbt]: Unfortunately, I'm not aware of any way to disable JS only for one Scala version (2.11 doesn't have any scala.js 1.0.0 artifact). One could do some black set magic to test Scala 2.11 with scala.js 1.0.0, but as these are only tests and all Scala versions are covered on the JVM, I would rather keep it simple like this.

File: build.sbt [hunk 2]
```diff
@@ -129,6 +134,7 @@ lazy val defaultScalacOptions = scalacOptions ++= Seq(
 lazy val defaultLibraryDependencies = libraryDependencies ++= Seq(
   scalaOrganization.value % "scala-reflect" % scalaVersion.value % Provided,
   scalaOrganization.value % "scala-compiler" % scalaVersion.value % Provided,
-  "org.scalacheck" %%% "scalacheck" % "1.14.0" % Test,
-  "org.scalatest" %%% "scalatest" % "3.0.8" % Test
+  "org.scalacheck" %%% "scalacheck" % "1.14.3" % Test,
+  "org.scalatest" %%% "scalatest" % "3.1.1" % Test,
+  "org.scalatestplus" %%% "scalacheck-1-14" % "3.1.1.0" % Test,
```
Comment (jhnsmth) [inline build.sbt:139]: The current version is `3.1.1.1`

Comment (ybasket) [reply build.sbt]: Just push an update for this (and Scala.js to 1.0.1), thanks!

**Comment by ybasket** [comment]
@carymrobbins may I ask you for a review?

File: build.sbt [hunk 3]
```diff
@@ -129,6 +134,7 @@ lazy val defaultScalacOptions = scalacOptions ++= Seq(
 lazy val defaultLibraryDependencies = libraryDependencies ++= Seq(
   scalaOrganization.value % "scala-reflect" % scalaVersion.value % Provided,
   scalaOrganization.value % "scala-compiler" % scalaVersion.value % Provided,
-  "org.scalacheck" %%% "scalacheck" % "1.14.0" % Test,
-  "org.scalatest" %%% "scalatest" % "3.0.8" % Test
+  "org.scalacheck" %%% "scalacheck" % "1.14.3" % Test,
+  "org.scalatest" %%% "scalatest" % "3.1.1" % Test,
+  "org.scalatestplus" %%% "scalacheck-1-14" % "3.1.1.1" % Test,
```
Comment (carymrobbins) [inline build.sbt:139]: Just curious, why was scalatestplus.scalacheck needed?

File: cats-tests/shared/src/test/scala/io/estatico/newtype/NewTypeCatsTest.scala [hunk 4]
```diff
@@ -4,10 +4,11 @@ import cats._
 import cats.implicits._
 import io.estatico.newtype.ops._
 import io.estatico.newtype.macros.{newsubtype, newtype}
-import org.scalatest.{FlatSpec, Matchers}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
 import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
 
-class NewTypeCatsTest extends FlatSpec with Matchers with ScalaCheckPropertyChecks {
+class NewTypeCatsTest extends AnyFlatSpec with Matchers with ScalaCheckPropertyChecks {
```
Comment (carymrobbins) [inline cats-tests/shared/src/test/scala/io/estatico/newtype/NewTypeCatsTest.scala:11]: How is `AnyFlatSpec` different? Is this just a change in the scalatest API?

File: build.sbt (hunk 1 above)
Comment (carymrobbins) [reply build.sbt:18]: Sounds good for now

File: build.sbt (hunk 3 above)
Comment (joroKr21) [reply build.sbt:139]: Scalacheck + Scalatest integration was factored out into another module

File: cats-tests/shared/src/test/scala/io/estatico/newtype/NewTypeCatsTest.scala (hunk 4 above)
Comment (joroKr21) [reply cats-tests/shared/src/test/scala/io/estatico/newtype/NewTypeCatsTest.scala:11]: Yes, `FlatSpec` is deprecated

**Comment by alexandru** [comment]
Hi all,

Can you please release a new version for Scala.js 1.x?
Eagerly waiting for it.

Thanks 🤗

**Comment by carymrobbins** [comment]
Thanks for your patience! I've cut the [v0.4.4 release](https://github.com/estatico/scala-newtype/releases/tag/v0.4.4) including this change.

**Pull request merged**